image/templates/helm/shared/templates/_scanner-v4_init.tpl.htpl

```diff
@@ -24,150 +24,114 @@
+{{- end -}}
+
 {{ $componentsCentralChart := dict "indexer" true "matcher" true }}
 {{ $componentsSecuredClusterChart := dict "indexer" true "matcher" false }}
 
-
 {{/* These will be propagated up. */}}
 {{ $components := dict "indexer" false "matcher" false }}
 {{ $dbEnabled  := false }}
 
-{{/* Handle high-level defaulting logic. */}}
 {{- if not $scannerV4Cfg.disable }}
+  {{/* Scanner V4 is switched on. */}}
 
-{{/* Sanity check. */}}
-{{- if not (or (eq $.Chart.Name "stackrox-central-services") (eq $.Chart.Name "stackrox-secured-cluster-services")) -}}
-  {{- include "srox.fail" (printf "Unexpected Helm chart name %q." $.Chart.Name) -}}
-{{- end -}}
-
-{{- if and (kindIs "invalid" $scannerV4Cfg.indexer.disable) (kindIs "invalid" $scannerV4Cfg.matcher.disable) -}}
-
-  {{/* Both settings scannerV4.indexer.disable and scannerV4.matcher.disable are not set by the user
-       => Pick default configuration based on the chart. */}}
+  {{/* Scanner V4 component configuration depends on the chart. */}}
   {{- if eq $.Chart.Name "stackrox-central-services" -}}
     {{- $components = $componentsCentralChart -}}
   {{- else -}}
     {{- $components = $componentsSecuredClusterChart -}}
   {{- end -}}
 
-{{- else -}}
-
-  {{/* The user has explicitly set either scannerV4.indexer.disable and/or scannerV4.matcher.disable.
-       Therefore, we use this explicitly set configuration. */}}
-  {{- if not (kindIs "invalid" $scannerV4Cfg.indexer.disable) -}}
-    {{- $_ = set $components "indexer" (not $scannerV4Cfg.indexer.disable) -}}
-  {{- end -}}
-  {{- if not (kindIs "invalid" $scannerV4Cfg.matcher.disable) -}}
-    {{- $_ = set $components "matcher" (not $scannerV4Cfg.matcher.disable) -}}
-  {{- end -}}
 
-  {{/* This is potentially an unsupported configuration. Hence we generate a warning. */}}
-  {{- $msg := printf "indexer is %s and matcher is %s." (ternary "enabled" "disabled" (get $components "indexer")) (ternary "enabled" "disabled" (get $components "matcher")) -}}
-  {{- if eq $.Chart.Name "stackrox-central-services" -}}
-    {{- if or (ne (get $components "indexer") (get $componentsCentralChart "indexer")) (ne (get $components "matcher") (get $componentsCentralChart "matcher")) -}}
-      {{- include "srox.warn" (list $ (printf "Unsupported Scanner V4 configuration detected: %s" $msg)) -}}
-    {{- end -}}
-  {{- else -}}
-    {{- if or (ne (get $components "indexer") (get $componentsSecuredClusterChart "indexer")) (ne (get $components "matcher") (get $componentsSecuredClusterChart "matcher")) -}}
-      {{- include "srox.warn" (list $ (printf "Unsupported Scanner V4 configuration detected: %s" $msg)) -}}
+  {{/* Configure images, certificates and passwords as required. */}}
+  {{ if get $components "indexer" }}
+    {{/* Note that for secured-cluster-services we don't configure certificates here,
+         instead they will be distributed at runtime by Sensor and Central. */}}
```

Comment:
I think this comment may make more sense a few lines down near ` {{/* Only generate certificate when installing central-services. */}}`

Reply:
done